### PR TITLE
make view logic reusable

### PR DESCRIPTION
**Pull request by Rizziepit**

### CODE REVIEW

**Comment by Rizziepit** [comment]
@smn @miltontony @nathanbegbie ready for review.

I need to add and override views in a springboard app. So I've split up the existing `SpringboardViews` class to make a reusable base class. The new structure:
- `springboard.views.base` with `SpringboardViews` base class
- `springboard.views.core` with `CoreViews` for category/pages/etc
- `springboard.views.auth` with `AuthViews` for login/logout

I'm happy to discard this PR if there's a nicer way to do this. But I'd like a way to reuse the context/elastic-git logic.

**Comment by Rizziepit** [comment]
Actually, reusability is just a part of it. I think we should start splitting things up into smaller chunks.

**Comment by smn** [comment]
Is it not possible to just override the url routing in the generated `application.py` to another view? Would that be a simpler approach perhaps?

**Comment by Rizziepit** [comment]
Yes, you'd have to override the url routing despite these changes.Have a look at how I'm using this: https://github.com/universalcore/springboard-iogt/pull/1. All I'm doing in this PR is reorganizing code so that there is a base class that view classes can extend. Maybe I'm misunderstanding you.

File: springboard/views/base.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        self.settings = request.registry.settings
+        es_host = self.settings.get('es.host', 'http://localhost:9200')
+        self.es_settings = {
+            'urls': [es_host]
+        }
+
+        repo_dir = self.settings.get('unicore.repos_dir', 'repos')
+        repo_names = map(
+            lambda repo_url: parse_repo_name(repo_url),
+            self.settings['unicore.content_repo_urls'].strip().split('\n'))
```
Comment (smn) [inline springboard/views/base.py]: We grew a `config_list()` function for this in [`springboard.utils`](https://github.com/universalcore/springboard/blob/develop/springboard/utils.py#L44).

**Pull request closed without merging**